tests/src/test/java/IOTests.java

```diff
@@ -3,29 +3,32 @@
 import mindustry.game.*;
 import mindustry.io.*;
 import org.junit.jupiter.api.*;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
 
 import java.nio.*;
 
 import static org.junit.jupiter.api.Assertions.*;
 
 public class IOTests{
 
-    @Test
-    void writeEnglish(){
-        ByteBuffer buffer = ByteBuffer.allocate(500);
-        TypeIO.writeString(buffer, "asd asd asd asd asdagagasasjakbgeah;jwrej 23424234");
-        buffer.position(0);
-        assertEquals(TypeIO.readString(buffer), "asd asd asd asd asdagagasasjakbgeah;jwrej 23424234");
-    }
-
-    @Test
-    void writeChinese(){
+    @ParameterizedTest
+    @ValueSource(strings ={
+            "asd asd asd asd asdagagasasjakbgeah;jwrej 23424234",
+            "这个服务器可以用自己的语言说话"
+    })
+    void writeStringTest(String string){
         ByteBuffer buffer = ByteBuffer.allocate(500);
-        TypeIO.writeString(buffer, "这个服务器可以用自己的语言说话");
+        TypeIO.writeString(buffer, string);
         buffer.position(0);
-        assertEquals(TypeIO.readString(buffer), "这个服务器可以用自己的语言说话");
+        assertEquals(TypeIO.readString(buffer), string);
     }
 
+    /**
```

Comment:
Any particular reason why you can't just update the libraries to 5.7 instead?

Reply:
No particular reason. I have updated the libraries